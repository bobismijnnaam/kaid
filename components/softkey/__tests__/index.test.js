import React from 'react';
import { shallow, render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import Softkey from '..';

configure({ adapter: new Adapter() });

describe('Softkey', function() {
  it('renders correctly', () => {
    const wrapper = shallow(<Softkey left="lsk" center="ok" right="rsk"/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders softkey buttons correctly', () => {
    const wrapper = shallow(<Softkey left="lsk" center="ok" right="rsk"/>);
    expect(toJson(wrapper.render())).toMatchSnapshot();
  });

  it('should have class "kai-softkey"', function() {
    expect(shallow(<Softkey />).is('.kai-softkey')).toBe(true);
  });

  it('should not able to render its children', function() {
    expect(render(<Softkey>softkey</Softkey>).text()).toEqual('');
  });
});
