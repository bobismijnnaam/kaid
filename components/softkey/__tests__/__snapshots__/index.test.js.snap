// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Softkey renders correctly 1`] = `
<form
  className="kai-softkey visible"
  data-type="action"
>
  <Button
    content="lsk"
    pos="left"
  />
  <Button
    content="ok"
    pos="center"
  />
  <Button
    content="rsk"
    pos="right"
  />
</form>
`;

exports[`Softkey renders softkey buttons correctly 1`] = `
<form
  class="kai-softkey visible"
  data-type="action"
>
  <button
    class="kai-softkey-btn"
    data-l10n-id="lsk"
    type="button"
  />
  <button
    class="kai-softkey-btn"
    data-l10n-id="ok"
    type="button"
  />
  <button
    class="kai-softkey-btn"
    data-l10n-id="rsk"
    type="button"
  />
</form>
`;
